### PR TITLE
Dont make checkbox required always

### LINKED ISSUE #67: Checkbox in array invalid when unchecked
In the refinement nodes like `mdref`there is the `mol_shape` parameter. 
This parameter is an array of booleans. Which is rendered by rjsf as list of checkboxes.

I am unable to submit a `mol_shape` array item which is unchecked. It gives an validation error. So I am unable to define that my second molecule should be treated as an shape, because I can not have the first array item (first molecule) unchecked. See screenshot below

localhost_3000_ (6)

### PR DESCRIPTION
**Pull request by sverhoeven**
Fixes #67

To test 
1. 1st way repeat steps in #67 
    1. Goto haddock3-download app at https://app.netlify.com/sites/wonderful-noether-53a9e8/deploys/62d930fd6466c80009103c7d
    2. Instead of using netlify hosted app you can also run locally from root of repo run `yarn && yarn dev`  and goto http://localhost:3000     
    3.  Add `mdref` node to workflow
    5. Expand molecule subform
    6. Press `+` button, this will add an unchecked checklbox
    7. Submit form
    8. Switch to text mode of workflow to see `mol_shape_1 = false`, before this PR it would have an invalid form.
1. 2nd way with story book
    1. from root of repo run `yarn storybook` 
    2. Goto http://localhost:6007
    4. In left sidebar expand `Checkbox`
    5. Select `Unchecked Checkbox In Array Is Submiteable` story
    6. Press `Submit` button, should not complain and in actions tab have a new entry with onSubmit[0].formData key have value `[false, true]`. The story presses the submit button when opening the story, so you will see 2 actions.
3. 3rd way with unit test
    1. Change to  `packages/form/` and run `yarn test`
    2. It should run `packages/form/src/IvresseCheckBoxWidget.test.tsx`and pass.

~This PR also adds vitest to form package and upgrades rjsf to latest version.~ Moved to #100

### CODE REVIEW

**Comment by sverhoeven** [comment]
This implementation blocks you from having a checkbox that must be checked to submit form. For example to agree to terms like a license.

File: packages/form/src/IvresseCheckboxWidget.stories.tsx [hunk 1]
```diff
@@ -41,3 +43,28 @@ export const CheckboxWidgetDescription: ComponentStory<typeof Form> = () => {
...
+      }
+    },
+    additionalProperties: false
+  }
+
+  return render(schema, { array: [false, true] })
+}
+
+export const UncheckedCheckboxInArrayIsSubmiteable = CheckboxArray.bind({})
+UncheckedCheckboxInArrayIsSubmiteable.play = async () => {
```
Comment (bvreede) [inline packages/form/src/IvresseCheckboxWidget.stories.tsx:68]: typo fix
```suggestion
export const UncheckedCheckboxInArrayIsSubmittable = CheckboxArray.bind({})
UncheckedCheckboxInArrayIsSubmittable.play = async () => {
```

**Review by bvreede — CHANGES_REQUESTED** [review]
Looks good! The bug in the workflow app has been fixed, the only thing I'm unsure about is the storybook; while the empty checkbox provided can be submitted without problems, any checkbox added does need to be ticked. Adding a checkbox then leaving it empty gives the error 

> .array[n] should be boolean

(where n is the number of the added checkbox).

+ small typo in the storybook :)

**Review by bvreede — APPROVED** [review]

**Comment by sverhoeven** [comment]
Thanks for reviewing

**Pull request merged**